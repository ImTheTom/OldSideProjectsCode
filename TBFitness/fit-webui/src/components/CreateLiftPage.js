import React from "react";
import Header from "./Header";
import axios from "axios";
import Config from "../config";
import { Input, Button } from "antd";

class CreateLiftPage extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			Name: "",
			Weight: "",
			Reps: "",
			Sets: ""
		};
		this.handleInputChange = this.handleInputChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleInputChange(event) {
		const target = event.target;
		const value = target.value;
		const name = target.name;

		this.setState({
			[name]: value
		});
	}

	handleSubmit(event) {
		event.preventDefault();
		axios.post(Config.host.NAME + "lift", {
			id: 0,
			user_id: 1,
			name: this.state.Name,
			weight: parseInt(this.state.Weight, 10),
			reps: parseInt(this.state.Reps, 10),
			sets: parseInt(this.state.Sets, 10),
		}).then(function (response) {
			console.log(response);
		}).catch(function (error) {
			console.log(error);
		});
	}

	render() {
		return (
			<div>
				<div>
					<Header />
				</div>
				<div className="CreateBox">
					<div className="create-lift-title">
						Create Lift
					</div>
					<form onSubmit={this.handleSubmit}>
						<Input className="create-lift-input" placeholder="Name" label="Name" name="Name" value={this.state.Name} onChange={(e) => this.handleInputChange(e)} /> <br></br>
						<Input className="create-lift-input" placeholder="Weight" label="Weight" name="Weight" value={this.state.Weight} onChange={(e) => this.handleInputChange(e)} /> <br></br>
						<Input className="create-lift-input" placeholder="Reps" label="Reps" name="Reps" value={this.state.Reps} onChange={(e) => this.handleInputChange(e)} /> <br></br>
						<Input className="create-lift-input" placeholder="Sets" label="Sets" name="Sets" value={this.state.Sets} onChange={(e) => this.handleInputChange(e)} /> <br></br>
						<Button className="create-lift-input" type="primary">Create Lift</Button>
					</form>
				</div>
			</div>
		);
	}
}


export default CreateLiftPage;
