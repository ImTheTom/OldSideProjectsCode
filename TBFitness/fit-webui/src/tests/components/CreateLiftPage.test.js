import CreateLiftPage from "../../components/CreateLiftPage";
import React from "react";
import { shallow } from "enzyme";

test("should render create Lift page correctly", () => {
	const wrapper = shallow(<CreateLiftPage />);
	expect(wrapper).toMatchSnapshot();
});
