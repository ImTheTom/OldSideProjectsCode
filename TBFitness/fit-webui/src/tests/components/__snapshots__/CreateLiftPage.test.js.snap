// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render create Lift page correctly 1`] = `
<div>
  <div>
    <Header />
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Name
        <input
          name="Name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        Weight
        <input
          name="Weight"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </label>
      <label>
        Reps
        <input
          name="Reps"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </label>
      <label>
        Sets
        <input
          name="Sets"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </label>
      <button>
        Create
      </button>
    </form>
  </div>
</div>
`;
